commands.js

```diff
@@ -137,7 +138,8 @@ var commands = exports.commands = {
 				var targetRoom = Rooms.search(targetRoomid) || Simulator.battles[targetRoomid];
 				if (!targetRoom) return connection.send('|pm|' + user.getIdentity() + '|' + targetUser.getIdentity() + "|/text The room '" + targetRoomid + "' does not exist.");
 				if (targetRoom.staffRoom && !targetUser.isStaff) return connection.send('|pm|' + user.getIdentity() + '|' + targetUser.getIdentity() + "|/text User '" + this.targetUsername + "' requires global auth to join room '" + targetRoom.id + "'.");
-				if (targetRoom.isPrivate && !((targetRoom.auth && targetRoom.auth[user.userid]) || targetUser.can('bypassall'))) {
+				if (targetRoom.isPrivate && targetRoom.modjoin &&
+						Config.groupsranking.indexOf(targetRoom.auth[targetUser.userid] || ' ') < Config.groupsranking.indexOf(targetRoom.modjoin) || !targetUser.can('bypassall')) {
```

Comment:
Not all users in the room with modchat up should be able to judge who should or shouldn't know about the room. If the invitee should be allowed to be invited, they'd either get promoted after approval from a room staff member or already have the rank required to join